### PR TITLE
[URGENT] Update flake.lock

### PR DESCRIPTION
**Pull request by GaetanLepage**
Automated changes by the [update-flake-lock](https://github.com/DeterminateSystems/update-flake-lock) GitHub Action.

```
Flake lock file updates:

• Updated input 'flake-parts':
    'github:hercules-ci/flake-parts/f7b3c975cf067e56e7cda6cb098ebe3fb4d74ca2' (2024-03-01)
  → 'github:hercules-ci/flake-parts/9126214d0a59633752a136528f5f3b9aa8565b7d' (2024-04-01)
• Updated input 'home-manager':
    'github:nix-community/home-manager/c0ef0dab55611c676ad7539bf4e41b3ec6fa87d2' (2024-03-28)
  → 'github:nix-community/home-manager/b787726a8413e11b074cde42704b4af32d95545c' (2024-04-06)
• Updated input 'nixpkgs':
    'github:NixOS/nixpkgs/d8fe5e6c92d0d190646fb9f1056741a229980089' (2024-03-29)
  → 'github:NixOS/nixpkgs/fd281bd6b7d3e32ddfa399853946f782553163b5' (2024-04-03)
• Updated input 'pre-commit-hooks':
    'github:cachix/pre-commit-hooks.nix/c11e43aed6f17336c25cd120eac886b96c455731' (2024-03-30)
  → 'github:cachix/pre-commit-hooks.nix/e35aed5fda3cc79f88ed7f1795021e559582093a' (2024-04-02)
```

### Running GitHub Actions on this PR

GitHub Actions will not run workflows on pull requests which are opened by a GitHub Action.

To run GitHub Actions workflows on this PR, run:

```sh
git branch -D update_flake_lock_action
git fetch origin
git checkout update_flake_lock_action
git commit --amend --no-edit
git push origin update_flake_lock_action --force
```

### CODE REVIEW

**Comment by GaetanLepage** [comment]
Let's wait for https://github.com/NixOS/nixpkgs/pull/301342 to land on `nixos-unstable` and then updating again.

**Comment by GaetanLepage** [comment]
I will merge this later today as it unblocks many people that wish to update their config.

**Comment by traxys** [comment]
If we need to fix urgenly things we should avoid doing work like switching to mkNeovimPlugin

**Comment by GaetanLepage** [comment]
> If we need to fix urgenly things we should avoid doing work like switching to mkNeovimPlugin

I agree, but for the two plugins in question, I didn't want to first introduce rename warnings to rewrite the plugin a few days later. I preferred to tackle this up front.

**Comment by traxys** [comment]
> > If we need to fix urgenly things we should avoid doing work like switching to mkNeovimPlugin
> 
> I agree, but for the two plugins in question, I didn't want to first introduce rename warnings to rewrite the plugin a few days later. I preferred to tackle this up front.

Yes but it makes a huge PR that is very awkard to review...

**Comment by traxys** [comment]
(In other words this is moving work from the author to the reviewer, and I feel this is _never_ a good compromise, because reviewing is often harder than writing something)

**Comment by GaetanLepage** [comment]
> > > If we need to fix urgenly things we should avoid doing work like switching to mkNeovimPlugin
> > 
> > 
> > I agree, but for the two plugins in question, I didn't want to first introduce rename warnings to rewrite the plugin a few days later. I preferred to tackle this up front.
> 
> Yes but it makes a huge PR that is very awkard to review...

Yes sorry... This update was massive. I am still waiting for more upstream fixes.

File: plugins/languages/debugprint.nix [hunk 1]
```diff
@@ -5,33 +5,111 @@
...
-      '';
+    maintainers = [maintainers.GaetanLepage];
+
+    # TODO introduced 2024-04-07: remove 2024-06-07
+    deprecateExtraOptions = true;
+    optionsRenamedToSettings = [
+      "moveToDebugline"
+      "displayCounter"
+      "displaySnippet"
+      "filetypes"
```
Comment (traxys) [inline plugins/languages/debugprint.nix:22]: We can't simply use a renamedToSettings, as the filetypes ooptions is an attrsOf option that uses `midVar` and `rightVar` as keys. We either need to  mark it removed, or do something more intelligent as compatibilty.

**Review by traxys — CHANGES_REQUESTED** [review]

File: plugins/languages/debugprint.nix (hunk 1 above)
Comment (GaetanLepage) [reply plugins/languages/debugprint.nix]: Indeed, done.

**Review by traxys — APPROVED** [review]

**Pull request merged**